ge/org.osate.ge/src/org/osate/ge/internal/ui/tools/CreateEndToEndFlowSpecificationTool.java

```diff
@@ -232,29 +246,18 @@ private Element getRefinedElement(final Element ce) {
 		return ce;
 	}
 
-	/**
-	 * Determines if subcomponent is valid context
-	 * @param context - context of the selected element
-	 */
-	private boolean isValidSubcomponent(final Context context) {
-		for (final Subcomponent subC : ci.getAllSubcomponents()) {
-			if (areEquivalent(subC, context)) {
-				return true;
-			}
-		}
-		return false;
-	}
-
 	private class CreateFlowsToolsDialog extends TitleAreaDialog {
 		private final NamingService namingService;
 		private final UiService uiService;
 		private final Aadl2Package pkg = Aadl2Factory.eINSTANCE.getAadl2Package();
 		private final EndToEndFlow eTEFlow = (EndToEndFlow) pkg.getEFactoryInstance().create(pkg.getEndToEndFlow());
-		private final List<EndToEndFlow> flows = new ArrayList<EndToEndFlow>();
 		private final List<String> segmentList = new ArrayList<String>();
 		private final List<String> modeList = new ArrayList<String>();
+
+		private Optional<EndToEndFlow> flow = Optional.empty();
```

Comment:
Unlike in other languages, it is usually not recommended to use `Optional` in fields. Replace with a regular field which may be null. ( I don't necessarily agree with this but it is standard java recommendation.)